diff --git a/solution-base/mongodb/charts/mongodb-sharded/templates/_helpers.tpl b/solution-base/mongodb/charts/mongodb-sharded/templates/_helpers.tpl
index cb923aaa..8f9587d7 100644
--- a/solution-base/mongodb/charts/mongodb-sharded/templates/_helpers.tpl
+++ b/solution-base/mongodb/charts/mongodb-sharded/templates/_helpers.tpl
@@ -37,6 +37,10 @@ Usage:
   {{- end -}}
 {{- end -}}
 
+{{- define "mongodb-sharded.configServer.serviceName" -}}
+  {{- printf "%s-configsvr.%s.svc.%s" (include "common.names.fullname" .) .Release.Namespace .Values.clusterDomain -}}
+{{- end -}}
+
 {{- define "mongodb-sharded.configServer.rsName" -}}
   {{- if .Values.configsvr.external.replicasetName -}}
     {{- .Values.configsvr.external.replicasetName }}
