diff --git a/solution-base/mongodb/charts/mongodb-sharded/templates/initialization-configmap.yaml b/solution-base/mongodb/charts/mongodb-sharded/templates/initialization-configmap.yaml
new file mode 100644
index 00000000..353971f6
--- /dev/null
+++ b/solution-base/mongodb/charts/mongodb-sharded/templates/initialization-configmap.yaml
@@ -0,0 +1,9 @@
+{{ if  (.Files.Glob "files/docker-entrypoint-initdb.d/*[sh|js|json]") }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: mongodb-sharded-init-scripts
+  labels: {{- include "common.labels.standard" . | nindent 4 }}
+data:
+{{ tpl (.Files.Glob "files/docker-entrypoint-initdb.d/*[sh|js|json]").AsConfig . | indent 2 }}
+{{ end }}
\ No newline at end of file
diff --git a/solution-base/mongodb/charts/mongodb-sharded/files/docker-entrypoint-initdb.d/create-app-user-sharded.sh b/solution-base/mongodb/charts/mongodb-sharded/files/docker-entrypoint-initdb.d/create-app-user-sharded.sh
new file mode 120000
index 00000000..61f367cb
--- /dev/null
+++ b/solution-base/mongodb/charts/mongodb-sharded/files/docker-entrypoint-initdb.d/create-app-user-sharded.sh
@@ -0,0 +1 @@
+../../../../scripts/create-app-user.sh
\ No newline at end of file
diff --git a/solution-base/mongodb/charts/mongodb-sharded/files/docker-entrypoint-initdb.d/set-default-write-concern-majority-sharded.sh b/solution-base/mongodb/charts/mongodb-sharded/files/docker-entrypoint-initdb.d/set-default-write-concern-majority-sharded.sh
new file mode 120000
index 00000000..9d7ca4a1
--- /dev/null
+++ b/solution-base/mongodb/charts/mongodb-sharded/files/docker-entrypoint-initdb.d/set-default-write-concern-majority-sharded.sh
@@ -0,0 +1 @@
+../../../../scripts/set-default-write-concern-majority.sh
\ No newline at end of file